http/README.md

```diff
@@ -1,172 +1,106 @@
 
 ```json
-{
-  "random": <multibase-encoded random bytes>,
-  "signature": <multibase-encoded signature>
-}
-```
-
-In order to keep this exchange stateless, the server SHOULD 1. include the current timestamp or an expiry data and 2. a signature in that data. This allows it to check in step 2 that it actually generated that data.
-The client MUST check that the signature obtained in the JSON response is correct and was generated using the same key that the server used to authenticate itself.
-
-The client signs the data received in step 1, and sends a POST request with the following JSON object to the server:
 
-```json
 {
-  "data": <the random bytes received from the server, multibase encoded>,
-  "peer-id": <peer ID, in string representation>,
-  "signature": <multibase-encoded signature>
+    "services": {
+        "/kad/1.0.0": "/kademlia/",
+        "/ipfs-http/1.0.0": "/",
+    }
 }
 ```
 
-The server verifies the signature and issues an authentication token. In order to allow stateless operation, at the very minimum, the authentication token SHOULD contain the peer ID. It SHOULD also contain an expiry date and it MAY be bound to the client’s IP address. The token is sent in the response body.
-
-The client uses the auth token on requests that require client authentication, by setting the `libp2p-auth-token` HTTP header.
-
-## Mapping to libp2p Streams
-
-libp2p services whose service is specified as request-response protocols can use a single protocol implementation to make the service available over HTTP as well as on top of libp2p streams.
-
-The libp2p protocol identifier is `/http1.1`. After negotiating this protocol using multistream-select, nodes treat the stream as a HTTP/1.1 stream for a single HTTP request (i.e. nodes MUST NOT use request pipelining).
+The resource contains a mapping of application protocols to their respective URL. For example, this configuration file would tell a client
 
-## Outlook: Interaction with Intermediaries
+1. That the Kademlia protocol is available at `/kademlia` and
+2. The [IPFS Path Gateway API](https://specs.ipfs.tech/http-gateways/path-gateway/) is mounted at `/`.
 
-One of the advantages of running HTTP is that there’s widely deployed caching infrastructure (CDNs). Content-addressed data is infinitely cacheable. Assuming a properly design data transfer protocol, retrieval for CIDs could be cached by the CDN and made available via a POP (geographically) close to the user, dramatically reducing retrieval latencies.
+It is valid expose a service at `/`. It is RECOMMENDED that the server resolve more specific URLs before less specific ones. e.g. a path of `/kademlia/foo` should be routed to the Kademlia protocol rather than the IPFS HTTP API.
```

Comment:
```suggestion
It is valid to expose a service at `/`. It is RECOMMENDED that the server resolve more specific URLs before less specific ones. e.g. a path of `/kademlia/foo` should be routed to the Kademlia protocol rather than the IPFS HTTP API.
```

Reply:
> It is RECOMMENDED that the server resolve more specific URLs before less specific ones

Do we need to specify this? Is there an RFC that deals with path resolution that we could link instead?